### ISSUE TITLE
Default Keras optimizers using XLA causes error with TensorFlow-DirectML plugin

### ISSUE THREAD

**Issue by maggie1059** [keras-team/tf-keras#337]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): win64, WSL
- TensorFlow installed from (source or binary): tensorflow-directml-plugin (PyPI)
- TensorFlow version (use command below): tensorflow-cpu==2.11
- Exact command to reproduce: python keras/engine/training_test.py

**Describe the problem**.

Due to the refactoring of optimizers in Keras 2.11, default optimizers created for Keras models using model.compile() default to using jit_compile=True, and this is only turned off if no GPU devices are found. However, DirectML does not support jit compile / XLA, and so the program will throw an error because it detects a GPU, even though the GPU doesn't support XLA.

**Describe the current behavior**.

The model will throw an error when the optimizer tries to [apply gradients](https://github.com/keras-team/keras/blob/a4ed7cbe53706535cc930e45ba0969afa7f89d71/keras/optimizers/optimizer.py#L1229):
`Could not find compiler for platform DML: NOT_FOUND: could not find registered compiler for platform DML -- was support for that platform linked in?`

**Describe the expected behavior**.

If a non-XLA-capable GPU is detected, optimizers/models should fall back to non-XLA instead of throwing an error. Currently, a workaround is to use the experimental optimizers instead, which have the option to specify jit_compile=False. However, the defaults should ideally work with TensorFlow plugins without having to use this kind of workaround.

**Standalone code to reproduce the issue**.

`pip install tensorflow-directml-plugin`
`python keras/engine/training_test.py`
